src/mongodb/scores.controller.ts

```diff
@@ -466,224 +474,1089 @@ export class ScoresController {
         }
       }
 
-      for (let responseTextELE of responseText.split("")) {
-        if (responseTextELE != ' ') {
-          if (vowelSignArr.includes(responseTextELE)) {
-            if (isPrevVowel) {
-              if (language !== 'hi') {
-                let prevEleArr = prevEle.split("");
-                prevEle = prevEleArr[0] + responseTextELE;
-                responseTokenArr.push(prevEle);
+
+      if (CreateLearnerProfileDto['contentType'].toLowerCase() !== "char") {
+        if (CreateLearnerProfileDto['output'] === undefined && CreateLearnerProfileDto.audio !== undefined) {
+          let audioFile = CreateLearnerProfileDto.audio;
+          const decoded = audioFile.toString('base64');
+          let audioOutput = await this.scoresService.audioFileToAsrOutput(decoded, CreateLearnerProfileDto.language);
+          CreateLearnerProfileDto['output'] = audioOutput.output;
+        }
+
+        let confidence_scoresArr = [];
+        let missing_token_scoresArr = [];
+        let anomaly_scoreArr = [];
+
+        responseText = CreateLearnerProfileDto.output[0].source;
+        let constructText = '';
+        let originalTextTokensArr = originalText.split("");
+        let responseTextTokensArr = responseText.split("");
+
+        let originalTextArr = originalText.split(" ");
+        let responseTextArr = responseText.split(" ");
+
+
+
+        let tokenHexcodeData = this.scoresService.gethexcodeMapping(language);
+        let tokenHexcodeDataArr = [];
+
+        await tokenHexcodeData.then((tokenHexcodedata: any) => {
+          tokenHexcodeDataArr = tokenHexcodedata;
+        });
+
+        // Prepare Constructed Text
+        let compareCharArr = [];
+
+        let constructTextSet = new Set();
+
+        let reptitionCount = 0;
+
+        for (let originalEle of CreateLearnerProfileDto.original_text.split(" ")) {
+          let originalRepCount = 0;
+          for (let sourceEle of responseText.split(" ")) {
+            if (similarity(originalEle, sourceEle) >= 0.40) {
```

Comment:
store the value of similarity and use it in the 2 if conditions, instead of calling the function twice.